src/hooks/useFavoriteCar.ts

```diff
@@ -0,0 +1,114 @@
+import { useEffect, useState } from 'react';
+import { GET_FAVORITE_CARS } from '../api/graphql/query/favoriteCars';
+import {
+  useDelete_User_CarsMutation,
+  useInsert_User_CarsMutation,
+  useQueryFavoriteCarsQuery,
+} from '../api/graphql/__generated__/graphql-types';
+import { useLoginContext } from '../context/LoginContext/LoginContext';
+
+const useFavoriteCar = (carId: number) => {
+  const [isFavorite, setIsFavorite] = useState<boolean>(false);

```

Comment:
Based on the initial value of type `boolean` you do not need to explicitly add it to the generic

Reply:
is redundant right? ok got it